src/app/pages/webinars/webinar-cards/webinar-grid-section.tsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import WebinarGrid from './webinar-grid';
+import {Webinar} from '../types';
+import './webinar-grid-section.scss';
+
+type WebinarFilter = (w: Webinar) => boolean;
+type Args = {
+    heading: string;
+    webinars: Webinar[];
+    filter?: WebinarFilter;
+    limit?: number;
+};
+
+export default function LabeledWebinarGrid({
+    heading,
+    webinars,
+    filter = () => true,
+    limit = 3,
+    children
+}: React.PropsWithChildren<Args>) {
+    const filteredWebinars = React.useMemo(
+        () => webinars.filter(filter).slice(0, limit),
+        [webinars, filter, limit]
+    );
+
+    if (webinars.length === 0) {
+        return null;
+    }
```

Comment:
could you instead do this check in the parent components and just not render this if webinars is empty?

Reply:
I could, but then it's a check in multiple places. I think it's better to have functions handle degenerate cases so the callers don't have to worry about it.